mount/mount.go

```diff
@@ -137,19 +135,19 @@ func (d *Datastore) Query(q query.Query) (query.Results, error) {
 			var more bool
 
 			for try := true; try; try = len(dses) > i {
-				if ds == nil {
+				if res == nil {
```

Comment:
just a personal preference that i should have caught in the previous PR, but we should probably add the `res != nil` check to the for-loop condition, since it essentially says "loop until res != nil or try is false.